### PR TITLE
Minor Refactoring

### PR DESCRIPTION
**Pull request by marcopax**
- Refactoring of some methods to use guards instead of creating a pyramid of doom.
- changed visibility of some class 
- use enum instead of fixed strings

### CODE REVIEW

File: MCCoreDataStack/Library/CoreDataStackManager/Extensions/MCCoreDataStackManager+Private.swift [hunk 1]
```diff
@@ -44,7 +43,7 @@ internal extension MCCoreDataStackManager {
 
             do {
                 if let storeCoordinator = self.storeCoordinator {
-                    var options = self.autoMigrationWithJournalMode("WAL")
+                    var options = self.autoMigrationWithJournalMode(CDJournalMode.WAL.rawValue)
```
Comment (marcocattai) [inline MCCoreDataStack/Library/CoreDataStackManager/Extensions/MCCoreDataStackManager+Private.swift:46]: :+1:

**Comment by marcocattai** [comment]
Can you change the destination of the pull request to Develop?

File: MCCoreDataStack/Library/CoreDataStackManager/MCCoreDataStackManager.swift [hunk 2]
```diff
@@ -70,112 +70,93 @@ public struct StackManagerHelper {
...
-            let delayTime = DispatchTime.now() + Double(Int64(0.2 * Double(NSEC_PER_SEC))) / Double(NSEC_PER_SEC)
-            DispatchQueue.main.asyncAfter(deadline: delayTime) {
-                if let completionBlockUnwrapped = completionBlock {
-                    completionBlockUnwrapped();
-                }
-            }
+    /// Configure the current CoreDataStack
+    /// - Parameter url: the URL of your sqlite file. See StackManagerHelper for help.
+    /// - Parameter configuration: configuration name
+    @objc open func configure(url: URL, configuration: String?, completion: MCCoreDataAsyncCompletion?) {
```
Comment (marcocattai) [inline MCCoreDataStack/Library/CoreDataStackManager/MCCoreDataStackManager.swift:107]: maybe it is better to change `url` to `storeURL` . At least it is more descriptive. If it was a `String` it was named as `storeUrl`

**Review by marcocattai — COMMENTED** [review]
See comments

File: MCCoreDataStack/Library/CoreDataStackManager/MCCoreDataStackManager.swift (hunk 2 above)
Comment (marcopax) [reply MCCoreDataStack/Library/CoreDataStackManager/MCCoreDataStackManager.swift:107]: Yeah I agree. I will change it with the next commit.

**Review by marcocattai — APPROVED** [review]

**Pull request merged**